src/status_im2/contexts/chat/messages/content/album/view.cljs

```diff
@@ -26,26 +27,37 @@
 
 (defn album-message
   [message]
-  [rn/view
-   {:style style/album-container}
-   (map-indexed
-    (fn [index item]
-      (let [images-count    (count (:album message))
-            images-size-key (if (< images-count 6) images-count :default)
-            size            (get-in constants/album-image-sizes [images-size-key index])]
-        [rn/view {:key (:message-id item)}
-         [fast-image/fast-image
-          {:style  (merge (style/image size index)
-                          {:border-top-left-radius     (border-tlr index)
-                           :border-top-right-radius    (border-trr index)
-                           :border-bottom-left-radius  (border-blr index images-count)
-                           :border-bottom-right-radius (border-brr index images-count)})
-           :source {:uri (:image (:content item))}}]
-         (when (and (> images-count max-display-count) (= index (- max-display-count 1)))
-           [rn/view
-            {:style (merge style/overlay {:border-bottom-right-radius (border-brr index images-count)})}
-            [quo/text
-             {:weight :bold
-              :size   :heading-2
-              :style  {:color colors/white}} (str "+" (- images-count 5))]])]))
-    (:album message))])
+  (let [shared-element-id (rf/sub [:shared-element-id])]
+    [rn/view
+     {:style style/album-container}
+     (map-indexed
+      (fn [index item]
+        (let [images-count    (count (:album message))
+              images-size-key (if (< images-count 6) images-count :default)
+              size            (get-in constants/album-image-sizes [images-size-key index])]
+          [rn/touchable-opacity
+           {:key            (:message-id item)
+            :active-opacity 1
+            :on-press       (fn []
+                              (rf/dispatch [:chat.ui/update-shared-element-id (:message-id item)])
+                              (js/setTimeout #(rf/dispatch [:chat.ui/navigate-to-horizontal-images
```

Comment:
[Light recommendation]

Not a fan of `js/setTimeout` calls here because we're trying to use the `utils.re-frame` ns to abstract some things. Similarly to the effect named `:utils/dispatch-later`, we could have an equivalent `rf/dispatch-later` function. The implementation is trivial.

See https://github.com/status-im/status-mobile/blob/develop/src/status_im/utils/utils.cljs#L86

Reply:
It still uses js/setTimeout tho, right?